Fix issue where sp_helpdb incorrectly rejects dbname parameter

raydhkim opened the pull request:
Previously, when sp_helpdb is executed with an optional database name as a parameter, Babelfish rejected the execution with an error saying that “The database ‘@dbname’ does not exist”, whereas sp_helpdb without any parameter works correctly.

Now, execution of sp_helpdb with a database name input works correctly, reporting information about the specified database, without throwing a wrong error message.

Task: BABEL-3539

Signed-off-by: Ray Kim <raydhkim@amazon.com>

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

raydhkim committed:
Fix issue where sp_helpdb incorrectly rejects dbname parameter

Previously, when sp_helpdb is executed with an optional database name as a parameter, Babelfish rejected the execution with an error saying that “The database ‘@dbname’ does not exist”, whereas sp_helpdb without any parameter works correctly.

Now, execution of sp_helpdb with a database name input works correctly, reporting information about the specified database, without throwing a wrong error message.

Task: BABEL-3539

Signed-off-by: Ray Kim <raydhkim@amazon.com>

raydhkim committed:
Fix issue where sp_helpdb incorrectly rejects dbname parameter

Previously, when sp_helpdb is executed with an optional database name as a parameter, Babelfish rejected the execution with an error saying that “The database ‘@dbname’ does not exist”, whereas sp_helpdb without any parameter works correctly.

Now, execution of sp_helpdb with a database name input works correctly, reporting information about the specified database, without throwing a wrong error message.

Task: BABEL-3539

Signed-off-by: Ray Kim <raydhkim@amazon.com>

xhfanhe commented on test/JDBC/expected/BABEL-sp_helpdb-vu-verify.out:
```diff
@@ -0,0 +1,60 @@
+GO
+~~START~~
+nvarchar#!#nvarchar#!#nvarchar#!#int#!#nvarchar#!#nvarchar#!#tinyint
+master#!#<NULL>#!#jdbc_user#!#1#!#2022-10-10 #!#<NULL>#!#<NULL>
```
I think it won't always succeed because the created date will change. We can create a table and insert the procedure output into it, like we did in sp_helpsrvrolemember

raydhkim committed:
Merge branch 'jira-babel-3539' of https://github.com/amazon-aurora/babelfish_extensions into jira-babel-3539

xhfanhe commented on test/JDBC/input/BABEL-sp_helpdb-vu-verify.sql:
```diff
@@ -0,0 +1,12 @@
+SELECT name, db_size, owner, status, compatibility_level FROM sys.babelfish_helpdb('msdb');
+GO
+SELECT name, db_size, owner, status, compatibility_level FROM sys.babelfish_helpdb('test_db1');
+GO
```
Please add a test for error case

xhfanhe replied:
Also you'll need to add BABEL-sp_helpdb into upgrade test schedules

xhfanhe commented on test/JDBC/input/BABEL-sp_helpdb-vu-prepare.sql:
```diff
@@ -0,0 +1,4 @@
+USE master
+
+CREATE DATABASE test_db1
```
Please change the name into <filename>_db, e.g. BABEL_sp_helpdb_db, so that when some error pumps up, we can easily locate the root cause.

raydhkim committed:
Merge remote-tracking branch 'upstream/BABEL_2_X_DEV' into jira-babel-3539

raydhkim committed:
Fix issue where sp_helpdb incorrectly rejects dbname parameter

Previously, when sp_helpdb is executed with an optional database name as a parameter, Babelfish rejected the execution with an error saying that “The database ‘@dbname’ does not exist”, whereas sp_helpdb without any parameter works correctly.

Now, execution of sp_helpdb with a database name input works correctly, reporting information about the specified database, without throwing a wrong error message.

Task: BABEL-3539

Signed-off-by: Ray Kim <raydhkim@amazon.com>

raydhkim committed:
Fix issue where sp_helpdb incorrectly rejects dbname parameter

Previously, when sp_helpdb is executed with an optional database name as a parameter, Babelfish rejected the execution with an error saying that “The database ‘@dbname’ does not exist”, whereas sp_helpdb without any parameter works correctly.

Now, execution of sp_helpdb with a database name input works correctly, reporting information about the specified database, without throwing a wrong error message.

Task: BABEL-3539

Signed-off-by: Ray Kim <raydhkim@amazon.com>

raydhkim committed:
Merge branch 'jira-babel-3539' of https://github.com/amazon-aurora/babelfish_extensions into jira-babel-3539

raydhkim committed:
Merge branch 'jira-babel-3539' of https://github.com/amazon-aurora/babelfish_extensions into jira-babel-3539

raydhkim committed:
Merge branch 'jira-babel-3539' of https://github.com/amazon-aurora/babelfish_extensions into jira-babel-3539

raydhkim committed:
Merge remote-tracking branch 'upstream/BABEL_2_X_DEV' into jira-babel-3539
